Add independent anchor+merkle checker

marcopeereboom opened the pull request:
This adds a tool `dcrtime_checker` that verifies a server proof with dcrdata without interacting with dcrtimed at all.

fernandoabolafio commented on cmd/dcrtime_checker/dcrtime_checker.go:
```diff
@@ -0,0 +1,227 @@
+		}
+
+		// Verify merkle path.
+		root, err = merkle.VerifyAuthPath(&v.ChainInformation.MerklePath)
```
I think we should have a step before to check the provided digest is part of the Merkle path, right? 
Otherwise, the proof can contain a valid Merkle path which doesn't have digest included in it.

marcopeereboom replied:
We do that. Read the comment right on top of the for loop on line 160.

fernandoabolafio commented on cmd/dcrtime_checker/dcrtime_checker.go:
```diff
@@ -0,0 +1,227 @@
+
+	return hex.EncodeToString(h.Sum(nil)), nil
+}
+
```
would be nice to have a quick explanation of what this function is doing

marcopeereboom replied:
These functions really need to move into util. Will do so.

fernandoabolafio commented on cmd/dcrtime_checker/dcrtime_checker.go:
```diff
@@ -0,0 +1,227 @@
+	var root *[32]byte
+	found := -1
+	for k, v := range vr.Digests {
+		if v.Digest != d {
```
Maybe extract the digest/proof validation code into another function? Just so the main function becomes more concise.

marcopeereboom replied:
We could but code is small enough that I feel it is not warranted.

marcopeereboom commented:
Note that travis is failing with a false positive at this time. Went over this with @davecgh and we both tried it locally and could not reproduce the travis error. Also note that the code that trips go vet did not change since the last commit.
